File: rb/features.rb
# BuddhaQuotes SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BuddhaQuotesFeatures
  def self.make_feature(name)
    case name
    when "base"
      BuddhaQuotesBaseFeature.new
    when "test"
      BuddhaQuotesTestFeature.new
    else
      BuddhaQuotesBaseFeature.new
    end
  end
end

File: py/features.py
# BuddhaQuotes SDK feature factory

from feature.base_feature import BuddhaQuotesBaseFeature
from feature.test_feature import BuddhaQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BuddhaQuotesBaseFeature(),
        "test": lambda: BuddhaQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
